bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/services/amp/AmpTransformer.java

```diff
@@ -97,6 +123,18 @@ public void startElement(String uri, String localName, String qName, Attributes
         contentHandler.startElement(uri, localName, qName, atts);
     }
 
+    @Override
+    public void characters(char[] ch, int start, int length) throws SAXException {
+
+        // Append the aggregated js to the head.
+        if (isAmp && !jsAppended) {
+            contentHandler.characters(jsContent.toCharArray(), 0, jsContent.length());
+            jsAppended = true;
+        }
+
+        contentHandler.characters(ch, start, length);
```

Comment:
won't this get weird with the indexes here? i could be wrong, but i would imagine that if you are appending the JS you should adjust the start and length params ( i could be wrong since you're calling the `.characters` twice). If all is rendering in the <head> correctly no matter the size of the JS libraries you're including, then I think it's fine. I've had to do this in the past to get it to work:
```    
public void characters(char[] ch, int start, int length) throws SAXException {
        String currentString = new String(ch);
        int placeLength = length;

        if (shouldRun()) {
            String currentPlace = currentString.substring(start, start + length);

            // Get the current placeholders in the string
            final List<String> placeholders = TemplateReplacementUtil.getPlaceholders(currentPlace);

            for (String placeholder : placeholders) {
                // Transform it to the key in the property map
                final String key = TemplateReplacementUtil.getKey(placeholder);

                // If the placeholder key is in the map then replace it
                if (properties.containsKey(key)) {
                    final String replaceValue = String.valueOf(properties.get(key));
                    final String replace = currentPlace.replace(placeholder, replaceValue);
                    placeLength = replace.length();
                    currentString = currentString.replace(currentPlace, replace);
                    currentPlace = replace;
                }
            }
        }

        getContentHandler().characters(currentString.toCharArray(), start, placeLength);
    }